crates/miden-tx/src/executor/notes_checker.rs

```diff
@@ -51,69 +71,68 @@ where
                     NoteAccountCompatibility::No => {
-                        // if the check failed, return a `Failure` with the vector of successfully
-                        // checked `P2ID` and `P2IDE` notes
-                        return Ok(NoteAccountExecution::Failure {
-                            failed_note_id: note.id(),
-                            successful_notes,
-                            error: None,
-                        });
+                        // If the check failed register the note as failed.
+                        failed.push(NoteConsumptionError::AccountCompatibilityError(
+                            note.into_note(),
+                        ));
                     },
 
-                    // this branch is unreachable, since we are handling the SWAP note separately,
+                    // This branch is unreachable, since we are handling the SWAP note separately,
                     // but as an extra precaution continue iterating over the notes and run the
-                    // transaction to make sure the note which returned "Maybe" could be consumed
-                    NoteAccountCompatibility::Maybe => continue,
+                    // transaction to make sure the note which returned "Maybe" could be consumed.
+                    NoteAccountCompatibility::Maybe => {
+                        maybe.push(note);
+                    },
                     NoteAccountCompatibility::Yes => {
-                        // put the successfully checked `P2ID` or `P2IDE` note to the vector
-                        successful_notes.push(note.id());
+                        // Put the successfully checked `P2ID` or `P2IDE` note to the vector.
+                        successful.push(note.into_note());
                     },
                 }
             } else {
-                // if we encountered not a well known note, then we have to execute the transaction
+                // If we encountered not a well known note, then we have to execute the transaction
                 // anyway, but we should continue iterating to make sure that there are no
-                // P2IDE notes which return a `No`
+                // P2IDE notes which return a `No`.
+                maybe.push(note);
                 continue;
             }
         }
 
-        // if all checked notes turned out to be either `P2ID` or `P2IDE` notes and all of them
-        // passed, then we could safely return the `Success`
-        if successful_notes.len() == (input_notes.num_notes() as usize) {
-            return Ok(NoteAccountExecution::Success);
+        // If all checked notes turned out to be either `P2ID` or `P2IDE` notes and all of them
+        // passed, then we could safely return the `Success`.
+        if successful.len() == input_note_count {
+            return Ok(NoteConsumptionInfo::new_successful(successful));
         }
```

Comment:
Similar to the above comment - I think we should remove this check as we always want to "dry run" the full set of notes.